Mathlib/Geometry/Manifold/ChartedSpace.lean

```diff
@@ -693,6 +694,24 @@ theorem ChartedSpace.locallyConnectedSpace [LocallyConnectedSpace H] : LocallyCo
   · rintro x s ⟨⟨-, -, hsconn⟩, hssubset⟩
     exact hsconn.isPreconnected.image _ ((e x).continuousOn_symm.mono hssubset)
 
+/-- If a topological space admits an atlas with locally path-connected charts, then the space itself
+  is locally connected. -/
+theorem ChartedSpace.locPathConnectedSpace [LocPathConnectedSpace H] : LocPathConnectedSpace M := by
+  refine ⟨fun x ↦ ⟨fun s ↦ ⟨fun hs ↦ ?_, fun ⟨u, hu⟩ ↦ Filter.mem_of_superset hu.1.1 hu.2⟩⟩⟩
+  let e := chartAt H x
+  let t := s ∩ e.source
+  have ht : t ∈ nhds x := Filter.inter_mem hs (chart_source_mem_nhds _ _)
+  have hts : t ⊆ s := inter_subset_left
+  use e.symm '' pathComponentIn (e x) (e '' t); refine ⟨⟨?_, ?_⟩, subset_trans ?_ hts⟩
```

Comment:
Small trick: you can combine `use` and `refine` into `refine ⟨e.symm '' pathComponentIn (e x) (e '' t), ⟨?_, ?_⟩, subset_trans ?_ hts⟩`